src/ImageSharp/Formats/Jpeg/Common/Block8x8F.Generated.cs

```diff
@@ -96,51 +96,139 @@ public void TransposeInto(ref Block8x8F d)
 		/// <summary>
         /// Level shift by +128, clip to [0, 255]
         /// </summary>
-        /// <param name="d">The destination block</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal void NormalizeColorsInto(ref Block8x8F d)
+        public void NormalizeColorsInplace()
         {
-            d.V0L = Vector4.Clamp(V0L + COff4, CMin4, CMax4);
-            d.V0R = Vector4.Clamp(V0R + COff4, CMin4, CMax4);
-            d.V1L = Vector4.Clamp(V1L + COff4, CMin4, CMax4);
-            d.V1R = Vector4.Clamp(V1R + COff4, CMin4, CMax4);
-            d.V2L = Vector4.Clamp(V2L + COff4, CMin4, CMax4);
-            d.V2R = Vector4.Clamp(V2R + COff4, CMin4, CMax4);
-            d.V3L = Vector4.Clamp(V3L + COff4, CMin4, CMax4);
-            d.V3R = Vector4.Clamp(V3R + COff4, CMin4, CMax4);
-            d.V4L = Vector4.Clamp(V4L + COff4, CMin4, CMax4);
-            d.V4R = Vector4.Clamp(V4R + COff4, CMin4, CMax4);
-            d.V5L = Vector4.Clamp(V5L + COff4, CMin4, CMax4);
-            d.V5R = Vector4.Clamp(V5R + COff4, CMin4, CMax4);
-            d.V6L = Vector4.Clamp(V6L + COff4, CMin4, CMax4);
-            d.V6R = Vector4.Clamp(V6R + COff4, CMin4, CMax4);
-            d.V7L = Vector4.Clamp(V7L + COff4, CMin4, CMax4);
-            d.V7R = Vector4.Clamp(V7R + COff4, CMin4, CMax4);
+            this.V0L = Vector4.Clamp(this.V0L + COff4, CMin4, CMax4);
+            this.V0R = Vector4.Clamp(this.V0R + COff4, CMin4, CMax4);
+            this.V1L = Vector4.Clamp(this.V1L + COff4, CMin4, CMax4);
+            this.V1R = Vector4.Clamp(this.V1R + COff4, CMin4, CMax4);
+            this.V2L = Vector4.Clamp(this.V2L + COff4, CMin4, CMax4);
+            this.V2R = Vector4.Clamp(this.V2R + COff4, CMin4, CMax4);
+            this.V3L = Vector4.Clamp(this.V3L + COff4, CMin4, CMax4);
+            this.V3R = Vector4.Clamp(this.V3R + COff4, CMin4, CMax4);
+            this.V4L = Vector4.Clamp(this.V4L + COff4, CMin4, CMax4);
+            this.V4R = Vector4.Clamp(this.V4R + COff4, CMin4, CMax4);
+            this.V5L = Vector4.Clamp(this.V5L + COff4, CMin4, CMax4);
+            this.V5R = Vector4.Clamp(this.V5R + COff4, CMin4, CMax4);
+            this.V6L = Vector4.Clamp(this.V6L + COff4, CMin4, CMax4);
+            this.V6R = Vector4.Clamp(this.V6R + COff4, CMin4, CMax4);
+            this.V7L = Vector4.Clamp(this.V7L + COff4, CMin4, CMax4);
+            this.V7R = Vector4.Clamp(this.V7R + COff4, CMin4, CMax4);
         }
 
-
         /// <summary>
-        /// Level shift by +128, clip to [0, 255]
+        /// AVX2-only variant for executing <see cref="NormalizeColorsInplace"/> and <see cref="RoundInplace"/> in one step.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal void NormalizeColorsInplace()
+        public void NormalizeColorsAndRoundInplaceAvx2()
+        {
+            Vector<float> off = new Vector<float>(128f);
```

Comment:
I know this doesn't appear to speed things up in the benchmarks but I think these should probably be static. It would reduce memory pressure and every little helps.

Reply:
@JimBobSquarePants I don't think there is a difference in this case.
A non-static, non-virtual method acts likes a static method taking `this` as the first argument. You should turn a method to `static` when you realize that you actually do not use `this` inside it. This is not the case here. (Or I'm just missing something?)

Reply:
Sorry, I mean the `Vector<T>` instance. I've seen similar elsewhere with `Vector4` also.

Reply:
The behaviour of JIT intrinsics is quite different to the behaviour we used to with normal data types / instructions.
- `new Vector<float>(128f)` will translate into a single CPU instruction initializing an AVX register from a constant
- `new Vector4(128f)` will do the same but with an SSE register AFAIK

If you make it static it will be ... well, actually what you gonna see depends on the quality of the JIT code generator. By my experience, for `Vector4` it tends to generate optimized code, but I never trust what it's doing with `Vector<T>`.

TL;DR: The simplest way to make sure that optimized code will be generated is to new it up, like I did 😄

Reply:
Ok, lets leave it as it is just now then. I'll let you do the merge 👍